pkg/s3select/sql/value.go

```diff
@@ -339,6 +340,52 @@ const (
 	opIneq = "!="
 )
 
+// InferBytesType will attempt to infer the data type of bytes.
+// Will fail if value type is not bytes or it would result in invalid utf8.
+// ORDER: int, float, bool, JSON, timestamp, string
+// If the content is valid JSON, the type will still be bytes.
+func (v *Value) InferBytesType() (err error) {
```

Comment:
@klauspost Where is this function used?

Reply:
It is actually unused.

Reply:
@donatello Correct. It seems like S3 always treats CSV values as "string". I guess I can just remove it.

Reply:
What we also wished to do was type inference and make our users life simple - this is something we diverged in our implementation..

Reply:
@harshavardhana  Yeah, that was my original thought, but it changes the output type, 

The tests would then give these results:
```
			query:      `SELECT num from s3object s WHERE id = 2`,
			wantResult: `{"num":"-5"}`,
->			wantResult: `{"num":-5}`,


			name:       "select-float-by-val",
			query:      `SELECT num2 from s3object s WHERE num2 = 0.765111`,
			wantResult: `{"num2":" 0.765111"}`,
->			wantResult: `{"num2":0.765111}`,

```
The string values are what S3 returns. So to be compatible I removed it - and thus why it is inactive. Of course values are still inferred for select comparisons, etc.

Reply:
@donatello @harshavardhana We have these options:

1) Keep values as strings, remove this code.
2) Keep values as strings, keep the code (it could easily be useful)
3) Convert values so values represent their duck type.

I have no particular preference, though I think we should be as compatible as possible.

Reply:
I think output being compatible with AWS is fine and also keeping the type inferred for comparisons is fine too.